File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/json/DateTypeAdapter.java
package com.elveselimoski.starterapp.json;

import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

import java.io.IOException;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class DateTypeAdapter extends TypeAdapter<Date> {

    private final static String DATE_FORMAT = "yyyy-MM-dd'T'hh:mmZ";

    public void write(JsonWriter out, Date value) throws IOException {
        out.value(value.toString());
    }

    public Date read(JsonReader jsonreader) throws IOException {
        String dateString = jsonreader.nextString();
        try {
            return new SimpleDateFormat(DATE_FORMAT, Locale.getDefault()).parse(dateString);
        } catch (ParseException e) {
            e.printStackTrace();
            return null;
        }
    }

}

File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/json/DateTypeAdapter.kt
package com.elveselimoski.starterapp.json

import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter

import java.io.IOException
import java.text.SimpleDateFormat
import java.util.*

class DateTypeAdapter : TypeAdapter<Date>() {
    @Throws(IOException::class)
    override fun write(out: JsonWriter, value: Date) {
        out.value(value.toString())
    }

    @Throws(IOException::class)
    override fun read(jsonreader: JsonReader): Date {
        val dateString = jsonreader.nextString()
        val date = SimpleDateFormat(DATE_FORMAT, Locale.getDefault()).parse(dateString)
        return date
    }

    companion object {
        private const val DATE_FORMAT = "yyyy-MM-dd'T'hh:mmZ"
    }
}
